### PR TITLE
fix(keychain): use filePath when uploading profile, #377

### LINKED ISSUE #377: [app] Public profile fetched from wrong url
My public profile is located at 

```
https://gaia.blockstack.org/hub/1Maw8BjWgj6MWrBCfupqQuWANthMhefb2v/0/profile.json
```
but `app` fetches it from 
```
https://gaia.blockstack.org/hub/1Maw8BjWgj6MWrBCfupqQuWANthMhefb2v/profile.json
```

Relevant code:
https://github.com/blockstack/ux/blob/master/packages/keychain/src/identity.ts#L111

### PR DESCRIPTION
**Pull request by friedger**
This PR
* fixes the write part of #377
* adds a filePath parameter to the uploadProfile

### CODE REVIEW

File: packages/keychain/src/profiles.ts [hunk 1]
```diff
@@ -59,30 +59,30 @@ export const registrars = {
...
-export function signProfileForUpload(profile: Profile, keypair: IdentityKeyPair) {
+export async function signProfileForUpload(
+  profile: Profile,
+  keypair: IdentityKeyPair
+): Promise<string> {
   const privateKey = keypair.key;
   const publicKey = keypair.keyID;
 
-  const token = signProfileToken(profile, privateKey, { publicKey });
+  const token = await signProfileToken(profile, privateKey, { publicKey });
```
Comment (hstove) [inline packages/keychain/src/profiles.ts]: @friedger the function signature for this in blockstack.js is not async?

File: packages/keychain/src/profiles.ts [hunk 2]
```diff
@@ -166,16 +171,18 @@ export const registerSubdomain = async ({
...
 }: {
   profile: Profile;
   gaiaHubUrl: string;
+  filePath: string;
   identity: Identity;
   gaiaHubConfig?: GaiaHubConfig;
 }) => {
-  const signedProfileTokenData = signProfileForUpload(profile, identity.keyPair);
-  await uploadProfile(gaiaHubUrl, identity, signedProfileTokenData, gaiaHubConfig);
+  const signedProfileTokenData = await signProfileForUpload(profile, identity.keyPair);
```
Comment (hstove) [inline packages/keychain/src/profiles.ts]: Also not async

**Review by hstove — CHANGES_REQUESTED** [review]
Thanks for this, Friedger. I'd love if you could add a test for this case. I can also write that if you'd like. We do need to change the signature for `signProfileForUpload`, and revert the changes to mark the `signProfile` function as async, as it's breaking a keychain test.

File: packages/keychain/src/profiles.ts (hunk 1 above)
Comment (friedger) [reply packages/keychain/src/profiles.ts]: In the profileToken.d.ts of blockstack in my local node_modules folder, the signProfileToken returns a Promise<string> and my IDE was complaining. Any ideas where this comes from?

Comment (friedger) [reply packages/keychain/src/profiles.ts]: @hstove I will revert the change, but I am still puzzled...

Comment (friedger) [reply packages/keychain/src/profiles.ts]: Looks like it was in an older version of blockstack.js. Resolved.

File: packages/keychain/src/profiles.ts (hunk 2 above)
Comment (friedger) [reply packages/keychain/src/profiles.ts]: 👍

**Comment by hstove** [comment]
Closing this - commits included in #578

**Pull request closed without merging**